mne/encoding/model.py

```diff
@@ -0,0 +1,238 @@
+            The output y array after removing rows based on ixs, then
+            based on the values in X. Only returned if `y` is passed.
+        """
+
+        # First remove indices if we've manually specified them
+        if y is not None:
+            if y.ndim == 1:
+                y = y[:, np.newaxis]
+        if ixs is not None:
+            if X.shape[0] < ixs.max():
+                raise ValueError('ixs exceed data shape')
+            X = X[ixs]
+            if y is not None:
+                y = y[ixs]
+
+        # Find data points with value we want to remove
+        if self.mask_val is not None:
+            if hasattr(self.mask_val, '__call__'):
+                # If a callable, pass X as an input, assume bool output
+                mask = self.mask_val(X)
+                if mask.ndim > 1:
+                    raise ValueError('Output mask must be of shape'
+                                     ' (n_samples,)')
+            else:
+                if self.mask_val is np.nan:
+                    mask = ~np.isnan(X)
+                else:
+                    mask = X != self.mask_val
+
+                # Decide which rows to remove
+                if self.mask_condition == 'all':
+                    # Remove rows w/ NO good values
+                    mask = mask.sum(axis=1) == 0
+                else:
+                    # Remove rows w/ AT LEAST 1 bad value
+                    mask = mask.sum(axis=1) < mask.shape[-1]
+
+            # Now change mask to rows that we wish to *keep*
+            self.mask = ~mask
+            mask_ixs = np.where(self.mask)[0]
+            mask_ixs = np.asarray(mask_ixs).squeeze()  # In case its a matrix
+            X = X[mask_ixs]
+            if y is not None:
+                y = y[mask_ixs]
+        else:
+            self.mask = np.ones(X.shape[0], dtype=bool)
+        # Return masked values
+        if y is not None:
+            return X, y
+        return X
+
+    def transform(self, X, y=None):
+        """Return a subset of data points.
+
+        Returns
+        -------
+        data : array, shape (n_masked_samples, n_features)
+            A subset of rows from the input array.
+        """
+        return self._mask_data(X, y)
+
+    @property
+    def _final_estimator(self):
+        return _get_final_est(self)
+
+
+def _get_final_est(est):
+    """Return the final component of a sklearn estimator/pipeline.
+
+    Parameters
+    ----------
+    est : pipeline, or sklearn / MNE estimator
+        An estimator chain from which you wish to pull the final estimator.
+
+    Returns
+    -------
+    est : the sklearn-style estimator at the end of the input chain.
+    """
+    # Define classes where we pull `est` manually
+    import sklearn
+    from distutils.version import LooseVersion
+    if LooseVersion(sklearn.__version__) < '0.16':
+        raise ValueError('Encoding models require sklearn version >= 0.16')
```

Comment:
I think it checks for sklearn most recent, or sklearn >= .15, but this needs sklearn >= 0.16 :P